internal/tiltfile/tiltfile_state.go

```diff
@@ -583,3 +591,10 @@ const (
 	claimPending
 	claimFinished
 )
+
+// A selector matches an entity if all non-empty selector fields match the corresponding entity fields
+type imageJSONPathSelector struct {
```

Comment:
ooh neat. do we want to generalize this to be a k8s selector rather than just an imageJsonPathSelector? (maybe leave a todo?)

Reply:
that's pretty much just a rename, unless I misunderstand? so I just did it

Reply:
yeah! i guess in future we could use this for like, `filter_yaml` and stuff? but no need to implement that now.